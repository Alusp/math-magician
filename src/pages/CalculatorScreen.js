import React from 'react';
import Calculator from '../components/Calculator';

function CalculatorScreen() {
  return (
    <div className="flex flex-row ">
      <div>
        <h1 className="container ml-14 text-2xl pt-32  font-semibold">Let us do some Maths!</h1>
      </div>
      <div className="container">
        <Calculator />
      </div>
    </div>
  );
}

export default CalculatorScreen;
