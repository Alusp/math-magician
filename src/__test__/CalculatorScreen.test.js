import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorScreen from '../pages/CalculatorScreen';
import '@testing-library/jest-dom';

describe('Test CalculatorScreen', () => {
  test('Testing CalculatorScreen', () => {
    const tree = renderer.create(<CalculatorScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
