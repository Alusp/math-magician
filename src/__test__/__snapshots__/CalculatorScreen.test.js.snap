// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CalculatorScreen Testing CalculatorScreen 1`] = `
<div
  className="flex flex-row "
>
  <div>
    <h1
      className="container ml-14 text-2xl pt-32  font-semibold"
    >
      Let us do some Maths!
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="flex justify-center items-center h-screen"
    >
      <div
        className="w-2/6 shadow-lg"
      >
        <div
          className="text-white bg-gray-500 flex justify-end p-5 w-full"
        >
          <span>
            0
          </span>
        </div>
        <table
          className="w-full"
        >
          <tbody
            className="w-2/6"
          >
            <tr>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  AC
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  +/-
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  %
                </button>
              </td>
              <td
                className="border hover:bg-gray-200 text-center bg-orange-500/90 text-xl"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  รท
                </button>
              </td>
            </tr>
            <tr>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  7
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  8
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  9
                </button>
              </td>
              <td
                className="border hover:bg-gray-200 text-center bg-orange-500/90 text-xl"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  x
                </button>
              </td>
            </tr>
            <tr>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  4
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  5
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  6
                </button>
              </td>
              <td
                className="border hover:bg-gray-200 text-center bg-orange-500/90 text-xl"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  -
                </button>
              </td>
            </tr>
            <tr>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  1
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  2
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  3
                </button>
              </td>
              <td
                className="border hover:bg-gray-200 text-center bg-orange-500/90 text-xl"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  +
                </button>
              </td>
            </tr>
          </tbody>
          <tfoot>
            <tr>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
                colSpan="2"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  0
                </button>
              </td>
              <td
                className="bg-white border hover:bg-gray-200 text-center"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  .
                </button>
              </td>
              <td
                className="border hover:bg-gray-200 text-center bg-orange-500/90 text-xl"
              >
                <button
                  className="p-6"
                  onClick={[Function]}
                  type="button"
                >
                  =
                </button>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
